crates/starknet/src/transactions/declare_transaction_v2.rs

```diff
@@ -20,6 +20,30 @@ pub struct DeclareTransactionV2 {
     pub chain_id: Felt,
 }
 
+impl DeclareTransactionV2 {
+    pub fn new(
+        sierra_contract_class: ContractClass,
+        compiled_class_hash: ClassHash,
+        sender_address: ContractAddress,
+        max_fee: u128,
+        signature: Vec<Felt>,
+        nonce: Felt,
+        chain_id: Felt,
+    ) -> Self {
+        Self {
+            sierra_contract_class,
+            compiled_class_hash,
+            sender_address,
+            max_fee,
+            signature,
+            nonce,
+            class_hash: None,
+            transaction_hash: None,
+            chain_id,
+        }
+    }
+}
+
 impl DeclareTransactionV2 {
```

Comment:
Should the two `impl` blocks be unified?

Reply:
changed in [71f0158](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/78/commits/71f01582630b8520e595a679da97e79c1e183fd3)